libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -96,6 +96,18 @@ void AP_AHRS_NavEKF::update(bool skip_ins_update)
         update_EKF3();
         update_EKF2();
     }
+
+    // update home location for GPS baro correction for DCM
+    Location EKF_home_loc;
+    if (_ekf_type == 2 && _ekf2_started && (EKF2.getOriginHgtMode() & (1<<2)) && EKF2.healthy()) {
+        EKF2.getOriginLLH(-1, EKF_home_loc);
```

Comment:
Right, isn't that exactly the case here? This is updating the height used if the vertical position is allowed to drift, however the DCM position isn't undergoing the same drift so I'm not sure why we would be taking this value to the DCM? (Highly probable I'm missing something about what this is trying to address)